src/java.desktop/unix/classes/sun/awt/screencast/TokenStorage.java

```diff
@@ -0,0 +1,410 @@
+        if (allowedScreenBounds == null) return;
+
+        TokenItem tokenItem = new TokenItem(newToken, allowedScreenBounds);
+
+        if (SCREENCAST_DEBUG) {
+            System.out.printf("// Storing TokenItem:\n%s\n", tokenItem);
+        }
+
+        synchronized (PROPS) {
+            String oldBoundsRecord = PROPS.getProperty(tokenItem.token, null);
+            String newBoundsRecord = tokenItem.dump();
+
+            boolean changed = false;
+
+            if (oldBoundsRecord == null
+                    || !oldBoundsRecord.equals(newBoundsRecord)) {
+                PROPS.setProperty(tokenItem.token, newBoundsRecord);
+                if (SCREENCAST_DEBUG) {
+                    System.out.printf(
+                            "// Writing new TokenItem:\n%s\n", tokenItem);
+                }
+                changed = true;
+            }
+
+            if (oldToken != null && !oldToken.equals(newToken)) {
+                // old token is no longer valid
+                if (SCREENCAST_DEBUG) {
+                    System.out.printf(
+                            "// storeTokenFromNative old token |%s| is "
+                                    + "no longer valid, removing\n", oldToken);
+                }
+
+                PROPS.remove(oldToken);
+                changed = true;
+            }
+
+            if (changed) {
+                store("save tokens");
+            }
+        }
+    }
+
+    private static boolean readTokens(Path path) {
+        if (path == null) return false;
+
+        try (BufferedReader reader = Files.newBufferedReader(path)) {
+            synchronized (PROPS) {
+                PROPS.clear();
+                PROPS.load(reader);
+            }
+        } catch (IOException e) {
+            if (SCREENCAST_DEBUG) {
+                System.err.printf("""
+                        Token storage: failed to load property file %s
+                        %s
+                        """, path, e);
+            }
+            return false;
+        }
+
+        return true;
+    }
+
+    static Set<TokenItem> getTokens(List<Rectangle> affectedScreenBounds) {
+        // We need an ordered set to store tokens
+        // with exact matches at the beginning.
+        LinkedHashSet<TokenItem> result = new LinkedHashSet<>();
+
+        Set<Map.Entry<Object, Object>> entries;
+        synchronized (PROPS) {
+            entries = PROPS.entrySet();
+        }
+
+        Set<String> malformed = new LinkedHashSet<>();
+
+        List<TokenItem> allTokenItems = entries
+                .stream()
+                .map(o -> {
+                    String token = String.valueOf(o.getKey());
+                    TokenItem tokenItem =
+                            TokenItem.parse(token, o.getValue());
+                    if (tokenItem == null) {
+                        malformed.add(token);
+                    }
+                    return tokenItem;
+                })
+                .filter(obj -> !Objects.isNull(obj))
```

Comment:
`Objects.isNull` wasn't supposed to be used like this. Let's either inline it and use `!= null` or use method reference `.filter(Objects::nonNull)`